api/v1alpha8/openstackmachine_types.go

```diff
@@ -91,6 +93,16 @@ type OpenStackMachineSpec struct {
 	IdentityRef *OpenStackIdentityReference `json:"identityRef,omitempty"`
 }
 
+type ServerMetadata struct {
+	// Key is the server metadata key
+	// kubebuilder:validation:MaxLength:=255
+	Key string `json:"key"`
+
+	// Value is the server metadata value
+	// kubebuilder:validation:MaxLength:=255
+	Value string `json:"value"`
+}
```

Comment:
I think we may have other K/V settings , maybe something like Metadata then others reuse it?

Reply:
ok, just check in case